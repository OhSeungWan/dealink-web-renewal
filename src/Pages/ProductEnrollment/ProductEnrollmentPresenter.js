import { Button, Container, ScreenWrapper } from 'Components/Atoms';
import { ImageBox, InfoList } from 'Components/Molecules';

import Header from 'Components/Molecules/Header';
import React from 'react';

const ProductEnrollmentPresenter = () => {
  return (
    <ScreenWrapper>
      <Container>
        <Header />
        <ImageBox type="upload" />
        <InfoList
          title="제목"
          content="로즈허브 빌드업 세럼+크림 set"
          horizon
          border
        />
        <InfoList
          title="제목"
          content="로즈허브 빌드업 세럼+크림 set"
          horizon
          border
        />
        <InfoList
          title="제목"
          content="로즈허브 빌드업 세럼+크림 set"
          horizon
          border
        />
        <InfoList title="제목" horizon border>
          <input></input>
        </InfoList>
        <InfoList title="제목" horizon border>
          <input></input>
        </InfoList>
        <InfoList title="제목" horizon border>
          <input></input>
        </InfoList>
        <Button primary common>
          상품등록
        </Button>
      </Container>
    </ScreenWrapper>
  );
};

export default ProductEnrollmentPresenter;
