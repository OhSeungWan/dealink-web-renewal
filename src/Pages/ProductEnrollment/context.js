import React from 'react';
export const ProductEnrollmentInput = {
  imageList: [],
  productTitle: '',
  productDetail: '',
  productPrice: '',
  kakaoUrl: '',
  d: '',
  h: '',
  m: '',
  s: ''
};

export const ProductEnrollmentContext = React.createContext(
  ProductEnrollmentInput
);
