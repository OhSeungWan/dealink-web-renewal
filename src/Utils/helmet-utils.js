import { Helmet } from 'react-helmet';
import React from 'react';
import { useLocation } from 'react-router-dom';
export default function HelmetMetaData(props) {
  let location = useLocation();
  let currentUrl = 'http://192.168.0.107:3000' + location.pathname;
  let quote = props.quote !== undefined ? props.quote : 'asfaf';
  let title =
    props.title !== undefined
      ? props.title
      : 'CampersTribe - World is yours to explore';
  let image =
    props.image !== undefined
      ? props.image
      : 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSmoBvNLxOeVV2QrGP_aoWv8nw2h4QwgQVbPw&usqp=CAU';
  let description =
    props.description !== undefined
      ? props.description
      : 'CampersTribe lets you experience the camping culture. We discover the hidden gems in the nearby to help you connect with nature & yourself by learning in the woods, on the riverbank under the open sky.' +
        'Trust us, its million dollars experience to ride away from city life, pitch a tent, do campfire and endless talk!' +
        'So, join us on this voyage, and explore the beauty and miracle of being yourself!';
  let hashtag = props.hashtag !== undefined ? props.hashtag : '#경매';
  return (
    <Helmet>
      <title>{title}</title>
      <meta charset="utf-8" />
      <meta name="csrf_token" content="" />
      <meta property="type" content="website" />
      <meta property="url" content={currentUrl} />
      <meta name="msapplication-TileColor" content="#ffffff" />
      <meta name="msapplication-TileImage" content="/ms-icon-144x144.png" />
      <meta name="theme-color" content="#ffffff" />
      <meta name="_token" content="" />
      <meta name="robots" content="noodp" />
      <meta property="title" content={title} />
      <meta property="quote" content={quote} />
      <meta name="description" content={description} />
      <meta property="image" content={image} />
      <meta property="og:locale" content="en_US" />
      <meta property="og:type" content="website" />
      <meta property="og:title" content={title} />
      <meta property="og:quote" content={quote} />
      <meta property="og:hashtag" content={hashtag} />
      <meta property="og:image" content={image} />
      <meta content="/*" property="og:image:type" />
      <meta property="og:url" content={currentUrl} />
      <meta property="og:site_name" content="CampersTribe" />
      <meta property="og:description" content={description} />{' '}
    </Helmet>
  );
}
