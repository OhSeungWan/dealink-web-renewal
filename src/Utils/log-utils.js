export const responselog = (name, res) => {
  console.log(`⇣-------------${name}-------------⇣`);
  console.log(res);
  console.log(`⇡_____________${name}_____________⇡`);
};
