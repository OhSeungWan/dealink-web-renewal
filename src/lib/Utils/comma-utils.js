export const comma = x => {
  if (typeof x == 'string') {
    return x.replace(/\B(?=(\d{3})+(?!\d))/g, ',');
  }
  return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
};
