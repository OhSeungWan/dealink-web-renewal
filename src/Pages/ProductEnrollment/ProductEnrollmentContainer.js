import React, { useEffect, useState } from 'react';
import { useHistory, useLocation } from 'react-router-dom';

import Date from 'Utils/date-utils';
import { Loading } from 'Components/Organisms/Modal';
import ProductEnrollmentPresenter from 'Pages/ProductEnrollment/ProductEnrollmentPresenter';
import { auctionApi } from 'Apis/auctionApi';
import { useProduct } from 'Hooks/useProduct';
import { useSelector } from 'react-redux';

const ProductEnrollmentContainer = () => {
  // TODO: refectorying
  const userInfo = useSelector(state => state.user);

  const location = useLocation();
  const [templink, setTempLink] = useState(location.state?.templink);

  //TODO: 리펙토링

  const history = useHistory();
  const [isOpen, setIsOpen] = useState(false);
  const [modalData, setModalData] = useState();
  const [value, setValue, loading, setLoading] = useProduct(templink);
  const [isTemp, setIsTemp] = useState(true);
  const valueValidate = value => {
    if (
      value.imageList.length == 0 ||
      value.productPrice == '' ||
      value.kakaoUrl == ''
    ) {
      setIsTemp(true);
      return true;
    } else {
      setIsTemp(false);
      return false;
    }
  };

  const closeModal = () => {
    setIsOpen(false);
    if (isTemp) {
      history.push(`/MyLink`, {
        before: true
      });
    } else {
      history.push(`/Product/seller/${userInfo.id}/${modalData.url}`, {
        before: true
      });
    }
  };

  const openModal = () => {
    setIsOpen(true);
  };

  const onSubmit = async () => {
    let form = new FormData();
    let now = new Date();
    if (isTemp) {
      alert('임시저장 됩니다.');
    }
    now.setDate(now.getDate() + parseInt(value.d));
    now.setHours(now.getHours() + parseInt(value.h));
    now.setMinutes(now.getMinutes() + parseInt(value.m));
    now.setSeconds(now.getSeconds() + parseInt(value.s));

    value.imageList.map(item => {
      form.append('productImages', item);
    });
    form.append(
      'auctionInfoRequest',
      JSON.stringify({
        startingPrice: value.productPrice.replace(/[^0-9]/g, ''), // 경매 시작가
        currentPrice: value.productPrice.replace(/[^0-9]/g, ''), // 경매 현재가
        closingTime: now.format('yyyy-MM-dd HH:mm:ss'), // 경매 마감시간
        tradingMethod: '직거래', //
        chatUrl: value.kakaoUrl, // 오픈채팅 주소
        name: value.productTitle,
        description: value.description,
        status: isTemp ? '6' : '0',
        url: templink ? templink.replace('/', '') : null
      })
    );
    setLoading(false);
    const data = await auctionApi.registerAuction(userInfo.id, {
      method: 'POST',
      headers: {
        AUTH_TOKEN: userInfo.accessToken
      },
      body: form
    });
    setModalData(data);
    setLoading(true);
    openModal();
  };

  useEffect(() => {
    valueValidate(value);
    console.log(value);
    console.log(isTemp);
  }, [value]);

  return loading ? (
    <ProductEnrollmentPresenter
      onSubmit={onSubmit}
      closeModal={closeModal}
      isOpen={isOpen}
      openModal={openModal}
      data={modalData}
      onChange={setValue}
      value={value}
      loading={loading}
      isTemp={isTemp}
      templink={templink}
      userInfo={userInfo}
      bannerType={location.pathname}
    />
  ) : (
    <Loading isOpen={true} />
  );
};

export default ProductEnrollmentContainer;
