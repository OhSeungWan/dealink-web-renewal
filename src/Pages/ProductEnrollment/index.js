import ProductEnrollmentContainer from 'Pages/ProductEnrollment/ProductEnrollmentContainer';
export default ProductEnrollmentContainer;
