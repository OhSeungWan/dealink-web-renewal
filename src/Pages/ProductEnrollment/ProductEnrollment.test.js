import { fireEvent, render, screen, waitFor } from 'Utils/test-utils';

import ProductEnrollment from 'Pages/ProductEnrollment';
import React from 'react';
import { mount } from 'enzyme';
import { rest } from 'msw';
import { setupServer } from 'msw/node';

// test 에서 사용할 서버와 Url을 생성

describe('<ProductEnrollment />', () => {
  // snapshot Test
  it('matches snapshot', () => {
    const wrapper = mount(<ProductEnrollment />);
    expect(wrapper).toMatchSnapshot();
  });

  // rendering Test
  it('renders', () => {
    const wrapper = mount(<ProductEnrollment />);
    const Header = wrapper.find('Header');
    const inputForm = wrapper.find('ImageBox');
    const submitButton = wrapper.find('button');
    const input = wrapper.find('input');

    expect(Header.length).toBe(1);
    //inputForm 1개
    expect(inputForm.length).toBe(1);
    //submitButton 1개
    expect(submitButton.length).toBe(1);
    //input 1개
    expect(input.length).toBe(7);
  });
});
