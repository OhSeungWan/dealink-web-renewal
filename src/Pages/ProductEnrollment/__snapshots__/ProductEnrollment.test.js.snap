// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductEnrollment /> matches snapshot 1`] = `ReactWrapper {}`;

exports[`<Profile /> matches snapshot 1`] = `ReactWrapper {}`;
